core/math/math_funcs.h

```diff
@@ -331,8 +331,8 @@ class Math {
 		}
 		// Then check for approximate equality.
 		float tolerance = CMP_EPSILON * abs(a);
```

Comment:
What about this one?

Reply:
Ah I forgot to mention I wasn't so sure about this one, it was deliberately missed out. Multiplying by `CMP_EPSILON` means the range here is potentially pretty small, and the extra accuracy could potentially make a difference. Worth doing but maybe in a separate PR just in case?

Reply:
Ended up doing some more testing of this, and I think we may be okay with it so I've added the `(float)` cast in the PR.